### PR TITLE
Update docs for `map_flatten` on `Option` case

### LINKED ISSUE #8345: map_flatten lint docs should include `Option` cases
### Description

Clippy suggests replacing `_.map(_).flatten(_)` with `.and_then(_)` for an `Option`. For example, this code:

```
Some(10)
  .map(|val| if val % 2 == 0 { Some(2) } else { None })
  .flatten();
```
generates this clippy lint
```
warning: called `map(..).flatten()` on an `Option`
 --> src/main.rs:2:13
  |
2 |       Some(10)
  |  _____________^
3 | |         .map(|val| if val % 2 == 0 { Some(2) } else { None })
4 | |         .flatten();
  | |__________________^ help: try using `and_then` instead: `.and_then(|val| if val % 2 == 0 { Some(2) } else { None })`
  |
  = note: `#[warn(clippy::map_flatten)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten
```
But the [webpage](https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten) linked in this lint suggests using `_.flat_map(_)` for `Option` types, yet `.flat_map(_)` does not exist on `Option`, instead the equivalent is `.and_then(_)`. This isn't a huge deal, but it can be confusing when the clippy suggests one thing, and the link with further information suggests something else.

### Version

```text
rustc 1.60.0-nightly (777bb86bc 2022-01-20)
binary: rustc
commit-hash: 777bb86bcdbc568be7cff6eeeaaf81a89b4aa50b
commit-date: 2022-01-20
host: x86_64-apple-darwin
release: 1.60.0-nightly
LLVM version: 13.0.0
```

### Additional Labels

_No response_

### PR DESCRIPTION
**Pull request by dswij**
closes #8345

changelog: [`map_flatten`] Add docs info for `Option` case

### CODE REVIEW

File: clippy_lints/src/methods/mod.rs [hunk 1]
```diff
@@ -566,17 +566,20 @@ declare_clippy_lint! {
...
     /// let vec = vec![vec![1]];
+    /// let opt = Some(5);
     ///
     /// // Bad
     /// vec.iter().map(|x| x.iter()).flatten();
+    /// opt.map(|x| Some(x*2)).flatten();
     ///
     /// // Good
     /// vec.iter().flat_map(|x| x.iter());
+    /// opt.and_then(|x|  Some(x*2));
```
Comment (giraffate) [inline clippy_lints/src/methods/mod.rs:582]: nits
```suggestion
    /// opt.and_then(|x| Some(x * 2));
```

File: clippy_lints/src/methods/mod.rs [hunk 2]
```diff
@@ -566,17 +566,20 @@ declare_clippy_lint! {
...
+    /// `_.flat_map(_)` for `Iterator` or `_.and_then(_)` for `Option`
     ///
     /// ### Example
     /// ```rust
     /// let vec = vec![vec![1]];
+    /// let opt = Some(5);
     ///
     /// // Bad
     /// vec.iter().map(|x| x.iter()).flatten();
+    /// opt.map(|x| Some(x*2)).flatten();
```
Comment (giraffate) [inline clippy_lints/src/methods/mod.rs:578]: nits
```suggestion
    /// opt.map(|x| Some(x * 2)).flatten();
```

**Review by giraffate — COMMENTED** [review]
LGTM with nits

**Comment by giraffate** [comment]
@bors r+

Thanks!

**Comment by ParkMyCar** [comment]
Thanks @dswij!

**Pull request merged**